src/enzyme_ad/jax/Dialect/EnzymeXLAOps.td

```diff
@@ -189,4 +189,17 @@ def CommRegionOp : EnzymeXLA_Op<"comm_region", [
   );
 }
 
+def LUFactorizationOp: EnzymeXLA_Op<"lu_factorization", [Pure]> {
```

Comment:
`Pure` is probably wrong here. LU routines always alias the input and output matrix

Reply:
no that's still right here. This can appear as pure (and is, returning a new tensor and taking in a tensor), even if the underlying implementation does things to generate it via a custom call